### PR TITLE
Replace ListView by RecyclerView.

### PR DESCRIPTION
**Pull request by Rajnish23**
Fixed - If an item is expanded random list items getting expanded.

## Description

Replace the Listview with RecyclerView.
Fixes - In CityHistoryAdapter if any Text is expanded randomly few items were getting expanded.

Fixes #(issue)

## Type of change
Just put an x in the [] which are valid.
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: Android/app/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#Thu Oct 01 11:48:10 AST 2020
```
Comment (Swati4star) [inline Android/app/gradle/wrapper/gradle-wrapper.properties:1]: revert this file

File: Android/app/gradlew [hunk 2]
```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env sh
```
Comment (Swati4star) [inline Android/app/gradlew:1]: remove this file

File: Android/app/gradlew.bat [hunk 3]
```diff
@@ -0,0 +1,84 @@
+@if "%DEBUG%" == "" @echo off
```
Comment (Swati4star) [inline Android/app/gradlew.bat:1]: remove this file too

Please add required changes only

**Review by Swati4star — CHANGES_REQUESTED** [review]

**Comment by Rajnish23** [comment]
Sure 👍

**Review by Rajnish23 — COMMENTED** [review]
Done with Required Changes.

**Comment by Swati4star** [comment]
There are still many gradle files which need to be reverted. Please see the "files changed" by this PR

**Comment by Rajnish23** [comment]
> There are still many gradle files which need to be reverted. Please see the "files changed" by this PR

There are total 7 files which are changed by this Pull Request. 4 of  them are CityHistoryActivity, CityHistoryAdapter, CityHistoryListItem and activity_city_history.xml. I have deleted the gradlew file and reverted the changes made in gradle-wrapper.properties. And the last one which is gradle-wrapper.jar.
I didn't made any changes to all other files in the project..

**Comment by Swati4star** [comment]
The gradle files  (these 3 files) are added by your PR which should be removed:
Android/app/gradle/wrapper/gradle-wrapper.jar
Android/app/gradle/wrapper/gradle-wrapper.properties
Android/app/gradlew.bat

**Comment by Rajnish23** [comment]
Done, Please Review it.

File: Android/app/gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#Mon Apr 29 12:39:08 CEST 2019
```
Comment (Swati4star) [inline Android/app/gradle/wrapper/gradle-wrapper.properties:1]: ??

Comment (Rajnish23) [reply Android/app/gradle/wrapper/gradle-wrapper.properties:1]: It's not added from my end.

Comment (Swati4star) [reply Android/app/gradle/wrapper/gradle-wrapper.properties:1]: it is

Comment (Rajnish23) [reply Android/app/gradle/wrapper/gradle-wrapper.properties]: #Mon Apr 29 12:39:08 CEST 2019
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-all.zip

This is what I found in master branch. Anyway I have removed that line. Please review it.

Comment (Swati4star) [reply Android/app/gradle/wrapper/gradle-wrapper.properties]: My comment wasn't about the line, it was about the file. I mentioned in my *previous comments* also to remove this file.

Anyways, I made a commit to remove that file.

**Review by Swati4star — APPROVED** [review]

**Comment by Rajnish23** [comment]
Thanks, Also please close the issue #822.

**Pull request merged**